### PR TITLE
Update version for 0.6.0 release.

### PR DESCRIPTION
**Pull request by MrBago**
Prepare the repo for a 0.6.0 release.

### CODE REVIEW

File: docs/quick-start.md [hunk 1]
```diff
@@ -30,15 +30,15 @@ We use the `--packages` argument to download the graphframes package and any dep
 <div data-lang="scala"  markdown="1">
 
 {% highlight bash %}
-$ ./bin/spark-shell --packages graphframes:graphframes:0.5.0-spark2.1-s_2.11
+$ ./bin/spark-shell --packages graphframes:graphframes:0.6.0-spark2.1-s_2.11
```
Comment (yogeshg) [inline docs/quick-start.md]: release notes say `2.3` support the version here seems off.

File: build.sbt [hunk 2]
```diff
@@ -23,7 +23,7 @@ name := "graphframes"
 spName := "graphframes/graphframes"
 
 // Don't forget to set the version
-version := s"0.6.0-SNAPSHOT-spark$sparkBranch"
+version := s"0.6.0-spark$sparkBranch"
```
Comment (yogeshg) [inline build.sbt]: should we bump `spark.version` in line 4 above?

File: README.md [hunk 3]
```diff
@@ -45,3 +45,7 @@ We welcome open source contributions as well!
   - Major bug fix: Indexing non-Integer vertex IDs, which is used by algorithms which call GraphX
     under the hood, including PageRank, ConnectedComponents, and others.
   - aggregateMessages for Python API
+- 0.6.0 release
+  - Spark 2.3 support
```
Comment (yogeshg) [inline README.md]: (dupe) seems inconsistent with build.sbt line 4

File: build.sbt (hunk 2 above)
Comment (MrBago) [reply build.sbt]: I think 2.3.1 is the latest public version. I don't think we can claim 2.4 support until it's officially released.

File: docs/quick-start.md (hunk 1 above)
Comment (MrBago) [reply docs/quick-start.md]: good catch.

**Comment by yogeshg** [comment]
offline discussion:
* spark version is already set to `2.3.1` in `build.sbt`
* `makefile` and `build.sbt` are in good state

LGTM pending explicit mention of 0.6 in pr header so it shows up in the squashed commit :)

**Comment by jkbradley** [comment]
Merging with master
Thanks @MrBago  and @yogeshg

**Pull request merged**